File: frontend/src/types/HazardTicket.ts
export interface HazardTicket {
  id: number;
  result_id: number;
  severity: string;
  owner_id: number;
  owner_name?: string | null;
  deadline: string;
  rectify_status: string;
  rectify_note: string;
  closed_at: string | null;
  created_at?: string | null;
  is_overdue?: boolean;
  device_id?: number | null;
  device_code?: string | null;
  device_type?: string | null;
  building_name?: string | null;
  item_code?: string | null;
  result_note?: string | null;
  measured_value?: string | null;
}

File: backend/src/models/hazard_ticket.py
from datetime import date, datetime

from sqlalchemy import Date, DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from src.config.database import Base


class HazardTicket(Base):
    __tablename__ = "hazard_ticket"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    result_id: Mapped[int] = mapped_column(ForeignKey("inspection_result.id"), nullable=False, index=True)
    severity: Mapped[str] = mapped_column(String(20), default="MEDIUM")
    owner_id: Mapped[int] = mapped_column(Integer, default=0)
    deadline: Mapped[date | None] = mapped_column(Date, nullable=True)
    rectify_status: Mapped[str] = mapped_column(String(20), default="OPEN")
    rectify_note: Mapped[str] = mapped_column(String(500), default="")
    closed_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.now)
